Resources/public/js/views/services/ez-roleserversideviewservice.js

```diff
@@ -0,0 +1,278 @@
+                            return;
+                        }
+                    });
+
+                if (error) {
+                    service._error('loadRole failed: ' + error.message);
+                    return;
+                }
+
+                service._assignRoleToEachAssignee(e, response.document.Role, end);
+
+                tasks.done(function () {
+                    service._assignRoleCallback(data.roleId, data.roleName, e.selection,
+                        Y.bind(data.afterUpdateCallback, service));
+                });
+            });
+        },
+
+        /**
+         * Assign the role to each user/group
+         *
+         * @method _assignRoleToEachAssignee
+         * @protected
+         * @param {EventFacade} e
+         * @param {Role} role
+         * @param {Function} end
+         */
+        _assignRoleToEachAssignee: function (e, role, end) {
+            var userService = this.get('capi').getUserService(),
+                service = this;
+
+            Y.Array.each(e.selection, function (struct) {
+                // TODO assign with limitation
+                var limitation = null,
+                    contentTypeIdentifier = struct.contentType.get('identifier');
+
+                //var limitation = { // DUMMY SECTION DATA
+                //    "_identifier": "Section",
+                //    "values": {
+                //        "ref": [{
+                //            "_href": "/api/ezp/v2/content/sections/1",
+                //            "_media-type": "application/vnd.ez.api.Section+json"
+                //        }, {
+                //            "_href": "/api/ezp/v2/content/sections/4",
+                //            "_media-type": "application/vnd.ez.api.Section+json"
+                //        }]
+                //    }
+                //};
+                //var limitation = { // DUMMY SUBTREE DATA
+                //    "_identifier": "Subtree",
+                //    "values": {
+                //        "ref": [{
+                //            "_href": "/api/ezp/v2/content/locations/1/5/14",
+                //            "_media-type": "application/vnd.ez.api.Location+json"
+                //        }]
+                //    }
+                //};
+
+                // Convert hrefs to IDs, required by API
+                if (limitation) {
+                    Y.Array.each(limitation.values.ref, function (ref, i, refs) {
+                        if (limitation._identifier == "Section") {
+                            refs[i]._href = /\d+$/.exec(ref._href)[0];
+                        } else if (limitation._identifier == "Subtree") {
+                            refs[i]._href = /(\/\d+)+$/.exec(ref._href)[0];
+                        }
+                    });
+                }
+
+                if (contentTypeIdentifier === 'user') {
+                    service._assignRoleToUser(userService.newRoleAssignInputStruct(role, limitation), struct, end);
```

Comment:
Maybe it would be easier to read if you would pass to the `_assignRoleToUser` method separately `role`, `limitation`, `content` (`struct.content`), `end` and build assignInputStruct inside that method.